hnn_core/optimization/metrics.py

```diff
@@ -0,0 +1,72 @@
+"""Metrics for parameter optimization."""
+
+# Authors: Carolina Fernandez <cxf418@miami.edu>
+#          Nick Tolley <nicholas_tolley@brown.edu>
+#          Ryan Thorpe <ryan_thorpe@brown.edu>
+#          Mainak Jas <mjas@mgh.harvard.edu>
+
+import numpy as np
+
+from hnn_core import simulate_dipole
+
+from scipy.signal import resample
+
+
+def _rmse_evoked(net, initial_params, set_params, predicted_params,
+                 update_params, obj_values, scale_factor, smooth_window_len,
+                 target, tstop):
+    """The objective function for evoked responses.
+
+    Parameters
+    ----------
+    net : Network
+        The network object.
+    initial_params : dict
+        Keys are parameter names, values are initial parameters.
+    set_params : func
+        User-defined function that sets network drives and parameters.
+    predicted_params : list
+        Parameters selected by the optimizer.
+    update_params : func
+        Function to update params.
+    scale_factor : float
+        The dipole scale factor.
+    smooth_window_len : float
+        The smooth window length.
+    tstop : float
+        The simulated dipole's duration.
+    target : ndarray
+        The recorded dipole.
+
+    Returns
+    -------
+    obj : float
+        Normalized RMSE between recorded and simulated dipole.
+    """
+
+    params = update_params(initial_params, predicted_params)
+
+    # simulate dpl with predicted params
+    new_net = net.copy()
+    set_params(new_net, params)
+    dpl = simulate_dipole(new_net, tstop=tstop, n_trials=1)[0]
+
+    # smooth & scale
+    dpl.scale(scale_factor)
+    if smooth_window_len is not None:
+        dpl.smooth(smooth_window_len)
+
+    # downsample if necessary
+    if (len(dpl.data['agg']) < len(target)):
+        target = resample(target, len(dpl.data['agg']))
+    elif (len(dpl.data['agg']) > len(target)):
+        dpl.data['agg'] = resample(dpl.data['agg'], len(target))
+
+    # calculate rmse
+    obj = np.sqrt(((dpl.data['agg'] - target)**2).sum() /
+                  len(dpl.times)) / (max(target) -
+                                     min(target))
```

Comment:
this computation looks common with `_rmse` @carolinafernandezp @rythorpe

Reply:
Oh yes, good catch.

Reply:
Also, what was the rationale @carolinafernandezp for normalizing the RMSE by `max(target) - min(target)`?

Reply:
@rythorpe When we first started drafting `_rmse_evoked` @jasmainak suggested we use the normalized RMSE so we're able to compare RMSEs between models of different scales, but please @jasmainak correct me if you meant something else I know there are a few different ways of calculating nRMSE (like by dividing by the mean of the data).

Reply:
I honestly have no recollection :) maybe it was @ntolley ? is the optimization sensitive to `scale_factor` ? how does that come into the picture?